data/src/dal/sqlite_dal.rs

```diff
@@ -2,134 +2,88 @@ use super::sqlite::{SqliteConnectionPool, SqliteDbConnectionError};
+        custom_path: String,
     ) -> Result<SqliteDal, SqliteDbConnectionError> {
-        let sqlite_db = SqliteConnectionPool::new(Some(directory)).await?;
-        Ok(SqliteDal { sql: sqlite_db })
+        let sqlite_db = SqliteConnectionPool::new(Some(custom_path)).await?;
+        Ok(SqliteDal {
+            sqlite_conn: sqlite_db,
+        })
     }
 
     pub async fn begin(&self) -> Result<Transaction<'_, Sqlite>, SqlTxError> {
-        match self.sql.pool.begin().await {
-            Ok(tx) => Ok(tx),
-            Err(e) => Err(SqlTxError::TxBegin(e)),
-        }
+        self.sqlite_conn
+            .pool
+            .begin()
+            .await
+            .map_err(|e| SqlTxError::TxBegin(e))
     }
 
+    /// Takes ownership of the given transaction object
     pub async fn rollback(&self, tx: Transaction<'_, Sqlite>) -> Result<(), SqlTxError> {
-        match tx.rollback().await {
-            Ok(_) => Ok(()),
-            Err(e) => Err(SqlTxError::TxRollback(e)),
-        }
+        tx.rollback().await.map_err(|e| SqlTxError::TxRollback(e))
     }
 
+    /// Takes ownership of the given transaction object
     pub async fn commit(&self, tx: Transaction<'_, Sqlite>) -> Result<(), SqlTxError> {
-        match tx.commit().await {
-            Ok(_) => Ok(()),
-            Err(e) => Err(SqlTxError::TxCommit(e)),
-        }
-    }
-
-    fn get_unix_timestamp(&self) -> i64 {
-        SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .expect("Time went backwards")
-            .as_secs() as i64
+        tx.commit().await.map_err(|e| SqlTxError::TxCommit(e))
     }
 
-    async fn execute_insert(
-        &self,
-        query: &str,
-        tx: Option<&mut Transaction<'_, Sqlite>>,
-    ) -> Result<i64, sqlx::Error> {
-        let query_result;
-        if let Some(trans) = tx {
-            query_result = sqlx::query(query).execute(&mut **trans).await?;
-        } else {
-            query_result = sqlx::query(query).execute(&self.sql.pool).await?;
-        }
-        Ok(query_result.last_insert_rowid())
+    /// Returns the current unix timestamp in seconds
+    fn get_unix_timestamp(&self) -> Result<i64, SqlQueryError> {
+        let duration_since_epoch = SystemTime::now().duration_since(UNIX_EPOCH)?;
+        Ok(duration_since_epoch.as_secs() as i64)
     }
 
-    async fn execute(
+    async fn execute_query(
         &self,
         query: &str,
         tx: Option<&mut Transaction<'_, Sqlite>>,
-    ) -> Result<(), sqlx::Error> {
-        if let Some(trans) = tx {
-            sqlx::query(query).execute(&mut **trans).await.map(|_| ())
-        } else {
-            sqlx::query(query).execute(&self.sql.pool).await.map(|_| ())
-        }
+    ) -> Result<SqliteQueryResult, sqlx::Error> {
+        let result = match tx {
+            Some(transaction) => sqlx::query(query).execute(&mut **transaction).await?,
+            None => sqlx::query(query).execute(&self.sqlite_conn.pool).await?,
+        };
```

Comment:
NIT: Usually recommended to _not_ use the `match` statement for an if/else case.
```suggestion
    let result = if let Some(transaction) = tx {
        sqlx::query(query).execute(&mut **transaction).await?
    } else {
        sqlx::query(query).execute(&self.sqlite_conn.pool).await?
    };
```